### ISSUE TITLE
"Delete share" reports share as empty

### ISSUE THREAD

**Issue by rayfoss** [rockstor/rockstor-core#2034]
I deleted a Share yesterday and got a popup asking me to verify. The popup stated the share had 0 bytes in size... so I didn't give it a second thought. Turns out, there was a terabyte of garbage in there.

My new share has over 500gb of stuff in it, yet the delete menu reports 0 bytes. This is worse than no size report at all.

![Screenshot from 2019-04-18 08-28-20](https://user-images.githubusercontent.com/876076/56364835-2341cb00-61b5-11e9-9065-e243cbbbbd17.png)

# Proposed Solution:
How about doing a quick `ls | wc -l` and saying about how many files are on the root directory?

**Comment by phillxnet** [comment]
@rayfoss Thanks for the report and agreed.

Could your indicate if you have quotas enabled (default) or not as with quotas disabled share size reports of 0 can be expected (as is indicated in our Web-UI quota enable/disable widgets): but obviously not welcome, and misleading, in this scenario.

Nice idea by the way re files count / alternative sources of into in this particular scenario.

Thanks again for highlighting this issue. Lets try and get a fix / work around for this soon.

@FroggyFlox Your ideas on this proposal, as always, would be welcome.

Far better for us to offer conflicting info ie 0 bytes is 340 files than what we present in the above scenario. And some better wording indicating that the 340 files would be deleted. Definitely need more  'play on the safer side' in this messaging.

We could even indicate here, given that we know if quotas are enabled on a given pool or not, contextually i.e. display a warning that data size is unreliable with quotas disabled for instance.

**Comment by FroggyFlox** [comment]
> saying about how many files are on the root directory

Hi @rayfoss, thanks for reporting this and for your proposed resolution. I agree with you and @phillxnet that showing the file count is a good alternative to indicate how much use the share has.


> We could even indicate here, given that we know if quotas are enabled on a given pool or not, contextually i.e. display a warning that data size is unreliable with quotas disabled for instance.

Agreed, I was thinking the same thing:
1. If quotas are enabled, show _x bytes and y files will be deleted_.
2. If quotas are **not** enabled, show _x bytes(i) and y files will be deleted_, with (i) referring to the tooltip icon explaining how share size is inaccurate when quotas are disabled.

@phillxnet, I was also wondering if we could refine point 2 above, but I may be completely wrong as btrfs share usage is still a tad unclear for me. Alternatively (or in addition to point 2 above), could we use a `du` report when quotas are disabled (in this situation at least)? It is still inaccurate but it may be preferred to show something closer to reality than `0 bytes`, pending all appropriate user warnings and tooltips.

**Comment by phillxnet** [comment]
@FroggyFlox Agreed 'du' as an option but it really has no idea on btrfs usage so it may just be redundant given we would have already indicated the 0 is inaccurate and currently our '0' reading tallies with our other tooltip re quotas disabled. It may just lead to bug reports indicating that it, in turn, is inaccurate. Lets leave that up to the implementer as we can't really win either way. At least until quotas is more reliable and faster.

So in summary we seem to have a consensus around 1. & 2. and add a du (but only on 2.) as you suggest but also indicate it to be unreliable. That way we cover our bases and give as much info as is available to us.

We really need quotas to be on all the time ideally given the complexity of identifying space in btrfs so it's a good thing there are performance improvements 'in the works' upstream.

@FroggyFlox & @rayfoss Thanks for your thoughts  / input.
